pandas/_libs/lib.pyx

```diff
@@ -1078,19 +1078,20 @@ cdef _try_infer_map(v):
     return None
 
 
-def infer_dtype(value: object, skipna: bool=False) -> str:
+def infer_dtype(value: object, skipna: bool=True) -> str:
     """
     Efficiently infer the type of a passed val, or list-like
     array of values. Return a string describing the type.
 
     Parameters
     ----------
     value : scalar, list, ndarray, or pandas type
-    skipna : bool, default False
-        Ignore NaN values when inferring the type. The default of ``False``
-        will be deprecated in a later version of pandas.
```

Comment:
Ugh, I don't know what "will be deprecated" means, but I don't know if it counts as a deprecations. A pre-deprecation warning :)

Unfortunately, this *is* exposed in the public API via `pandas.api.types`. http://pandas-docs.github.io/pandas-docs-travis/generated/pandas.api.types.infer_dtype.html

How much of a hassle is it for you do change this to `skipna=None`, warning when it's None, and setting the value to `False` for now (switching to True in the next version)?